### PR TITLE
72 us 27 admin merchant enabledisable 2

### PR DESCRIPTION
**Pull request by s2an**
Done. A bit odd maybe, but done. Hopefully it is easy to merge the conflicts.

### CODE REVIEW

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -7,18 +7,49 @@ def show
...
   def update
     merchant = Merchant.find(params[:id])
-    if merchant.update(name:params[:"Merchant Name"])
-      redirect_to admin_merchant_path(merchant)
+
+    if merchant.update(admin_merchants_params)
       flash[:success] = "Information has succesfully been updated"
+
+      if params[:merchant] && params[:merchant][:status]
+        redirect_to admin_merchants_path
```
Comment (Dylan-Perry) [inline app/controllers/admin/merchants_controller.rb:21]: Great use of the route prefixes here Steve, yer a legend

File: app/views/admin/merchants/index.html.erb
Comment (Dylan-Perry) [inline app/views/admin/merchants/index.html.erb:1]: This is extremely clean. Great passage of value/parameter using the value key

File: db/migrate/20240110041931_add_status_to_merchant.rb
Comment (Dylan-Perry) [inline db/migrate/20240110041931_add_status_to_merchant.rb:1]: Fixing forward, love it

File: spec/factories/merchant.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 FactoryBot.define do
   factory :merchant do
     name { Faker::Name.name  }
+    status { "" }
```
Comment (Dylan-Perry) [inline spec/factories/merchant.rb:4]: I would not have thought to update the factories with the new field. Smart

File: app/controllers/admin/merchants_controller.rb [hunk 3]
```diff
@@ -7,18 +7,49 @@ def show
...
+
   def update
     merchant = Merchant.find(params[:id])
-    if merchant.update(name:params[:"Merchant Name"])
-      redirect_to admin_merchant_path(merchant)
+
+    if merchant.update(admin_merchants_params)
       flash[:success] = "Information has succesfully been updated"
+
+      if params[:merchant] && params[:merchant][:status]
```
Comment (megroth2) [inline app/controllers/admin/merchants_controller.rb:20]: very minor and I don't think we need to change this due to the time crunch and if tests pass, but would simplifying this line to `if params[:merchant][:status]` work as well? I could be missing something, but just curious if that works also

**Review by Dylan-Perry — APPROVED** [review]

**Review by megroth2 — APPROVED** [review]

File: app/controllers/admin/merchants_controller.rb (hunk 3 above)
Comment (Dylan-Perry) [reply app/controllers/admin/merchants_controller.rb:20]: That is a good point, I also imagine it would. Great callout

**Pull request merged**